### PR TITLE
feat: ledger signing support

### LINKED ISSUE #6831: [Task]: Add capability to sign transaction with ledger
### Task description

We need a way to sign transactions with the ledger ethereum app.

### Requirements

TBD

### Creation checklist

- [ ] I have assigned this task to the correct people
- [X] I have added the most appropriate labels
- [X] I have linked the correct milestone and/or project

### PR DESCRIPTION
**Pull request by Tuditi**
## Summary

closes #6831 

...

## Changelog

```
- Added ledger signing function signTransactionData()
- Added buildBip32Path() and deconstructBip32Path()
- Added tests
```

### Platforms

> Please select any platforms where your changes have been tested.

-   **Desktop**
    -   [ ] MacOS
    -   [x] Linux
    -   [ ] Windows
-   **Mobile**
    -   [ ] iOS
    -   [ ] Android

### Instructions

> Please describe the specific instructions, configurations, and/or test cases necessary to **test and verify** that your changes work as intended.

...

## Checklist

> Please tick the following boxes that are relevant to your changes.

-   [ ] I have followed the contribution guidelines for this project
-   [ ] I have performed a self-review of my own code
-   [ ] I have commented my code, particularly in hard-to-understand areas
-   [ ] I have added or modified tests that prove my changes work as intended
-   [ ] I have verified that new and existing unit tests pass locally with my changes
-   [ ] I have verified that my latest changes pass CI workflows for testing and linting
-   [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: packages/shared/lib/core/layer-2/actions/getSelectedAccountLayer2Tokens.ts [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import { getSelectedAccountWrappedNativeTokens } from './getSelectedAccountWrappedNativeTokens'
 
-export async function getSelectedAccountLayer2Tokens(): Promise<{ amount: bigint; id: string }[]> {
+export async function getSelectedAccountLayer2Tokens(): Promise<unknown[]> {
```
Comment (MarkNerdi) [inline packages/shared/lib/core/layer-2/actions/getSelectedAccountLayer2Tokens.ts:3]: are the changes in this file some merge issues? 
i commented on some of matts PRs to revert these changes, and he did. So maybe these are some artifacts by branching off of that branch

File: packages/desktop/electron/lib/ledger.js [hunk 2]
```diff
@@ -10,34 +10,91 @@ const TransportNodeHid = require('@ledgerhq/hw-transport-node-hid').default
...
     return { evmAddress: data.address, coinType, accountIndex }
 }
 
 function buildBip32Path(coinType, accountIndex) {
     return `44'/${coinType}'/${accountIndex}'/0/0`
 }
+
+async function signTransactionData(data, coinType, accountIndex) {
+    const appEth = new AppEth(transport)
+    const common = new Common({ chain: Chain.Sepolia })
```
Comment (MarkNerdi) [inline packages/desktop/electron/lib/ledger.js:75]: use shimmerEVM

File: packages/shared/lib/core/app/interfaces/evm-address.interface.ts [hunk 3]
```diff
@@ -1,4 +1,4 @@
-export interface IEvmAddressEvent {
+export interface IEvmAddress {
```
Comment (MarkNerdi) [inline packages/shared/lib/core/app/interfaces/evm-address.interface.ts:1]: IMO the `event` suffix makes it more clear that this is only the returned event when we receive it

File: packages/shared/lib/core/layer-2/utils/prepareErc20EvmTransactionData.ts [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+import { ERC20_ABI } from '../abis'
+import { GAS_MULTIPLIER } from '../constants'
+import { IErc20EvmTransactionOptions, IEvmTransactionData } from '../interfaces'
+
+export async function prepareErc20EvmTransactionData(
+    options: IErc20EvmTransactionOptions
+): Promise<IEvmTransactionData> {
+    const { tokenAddress, originAddress, recipientAddress, transferAmount } = options
+
+    const provider = get(network)?.getChain(1071)?.getProvider()
```
Comment (MarkNerdi) [inline packages/shared/lib/core/layer-2/utils/prepareErc20EvmTransactionData.ts:14]: use constant or enum instead of magic number

**Review by MarkNerdi — CHANGES_REQUESTED** [review]
Some small changes

File: packages/shared/lib/core/layer-2/actions/getSelectedAccountLayer2Tokens.ts (hunk 1 above)
Comment (Tuditi) [reply packages/shared/lib/core/layer-2/actions/getSelectedAccountLayer2Tokens.ts]: Yup, I fixed them.

File: packages/desktop/electron/lib/ledger.js (hunk 2 above)
Comment (Tuditi) [reply packages/desktop/electron/lib/ledger.js]: I removed showing the Chain Id, here because it requires more work. You have to config a custom chain and we're lacking some data.

**Review by MarkNerdi — APPROVED** [review]
LGTM
